src/index.ts

```diff
@@ -69,18 +68,26 @@ function getCurrentTraceFromAgent() {
 export class LoggingBunyan extends Writable {
   private logName: string;
   private resource: types.MonitoredResource|undefined;
-  private serviceContext: types.ServiceContext|undefined;
+  private serviceContext: types.ServiceContext;
   stackdriverLog:
       types.StackdriverLog;  // TODO: add type for @google-cloud/logging
   constructor(options?: types.Options) {
     options = options || {};
     super({objectMode: true});
     this.logName = options.logName || 'bunyan_log';
     this.resource = options.resource;
-    this.serviceContext = options.serviceContext;
+    this.serviceContext = options.serviceContext || {};
     this.stackdriverLog = logging(options).log(this.logName, {
       removeCircular: true,
     });
+
+    // serviceContext.service is required by the Error Reporting
+    // API.  Without it, errors that are logged with level 'error'
+    // or higher will not be displayed in the Error Reporting
+    // console
+    if (!this.serviceContext.service) {
+      this.serviceContext.service = 'default';
```

Comment:
I left a comment earlier but I don't see it anywhere.

If the user *intended* to provide the service context, but it incorrectly, we should throw.

Reply:
The other comment I had added was.. is there precent from other languages?